src/api/java/com/minecolonies/api/colony/workorders/IWorkOrder.java

```diff
@@ -34,26 +42,129 @@
      */
     void setPriority(int priority);
 
+    /**
+     * Get the structure this work order should be using, if any.
+     *
+     * @return the schematic name.
+     */
+    String getStructureName();
+
+    /**
+     * Get the current level of the structure of the work order.
+     *
+     * @return the current level.
+     */
+    int getCurrentLevel();
+
+    /**
+     * Get the target level of the structure of the work order.
+     *
+     * @return the target level.
+     */
+    int getTargetLevel();
+
+    /**
+     * Get the amount of resources this work order still requires.
+     *
+     * @return the amount of resources.
+     */
+    int getAmountOfResources();
+
+    /**
+     * Set the amount of resources this work order still requires.
+     *
+     * @param newQuantity the new amount of resources.
+     */
+    void setAmountOfResources(int newQuantity);
+
+    /**
+     * Get the iterator type (method of construction) of the work order.
+     *
+     * @return the iterator type.
+     */
+    String getIteratorType();
+
+    /**
+     * Set the iterator type (method of construction) of the work order.
+     *
+     * @param iteratorType the new iterator type.
+     */
+    void setIteratorType(String iteratorType);
+
+    /**
+     * Whether the area of the work order has been cleared out or not.
+     *
+     * @return true if the area is cleared.
+     */
+    boolean isCleared();
+
+    /**
+     * Set whether the area of the work order has been cleared out or not.
+     *
+     * @param cleared the new cleared state.
+     */
+    void setCleared(boolean cleared);
+
+    /**
+     * Set whether the resources for this work order have been requested.
+     *
+     * @return true when the resources are requested.
+     */
+    boolean isRequested();
+
+    /**
+     * Set whether the building has been cleared.
+     *
+     * @param requested true if the building has been cleared.
+     */
+    void setRequested(boolean requested);
+
     /**
      * Checks if the workOrder has changed.
      *
      * @return true if so.
      */
-    boolean hasChanged();
+    boolean isChanged();
```

Comment:
could you rename this to isDirty()? we're using that everywhere else with markDirty()

Reply:
I'm gonna double check if this even does anything important at all because the internal `changed` variable doesn't even seem to update on every setter (not previously and not currently), so it might just be redundant.